### PR TITLE
Trigger AWS Lambda tests on label

### PR DESCRIPTION
**Pull request by sentrivana**
Our AWS Lambda test suite currently doesn't run properly on external contributor PRs because it needs access to repo secrets, which it currently doesn't have. This PR adds a label to grant access to the secrets, which is invalidated upon any new code changes.

## How it works

For the AWS Lambda test suite (and any future test suites that need access to GH secrets):
* add a new `check-permissions` job that runs before all test jobs
  * the `check-permissions` job runs a Python script (taken from [here](https://github.com/getsentry/sentry/blob/master/.github/workflows/scripts/getsentry-dispatch-setup)) that finishes with an error code if the PR was not made by someone with write permissions *and* there is no `Trigger: tests using secrets` label on the PR
  * additionally, the job removes the `Trigger: tests using secrets` label on any code changes before any code from the PR is checked out
* make all test jobs depend on `check-permissions` finishing successfully
* run the workflow on `pull_request_target` (with access to secrets)

Copied and adapted the approach from `sentry`: https://github.com/getsentry/sentry/blob/master/.github/workflows/getsentry-dispatch.yml

The test AWS account has been stripped down of all unnecessary permissions: https://github.com/getsentry/sentry-python/pull/2493#issuecomment-1809784497

Since the workflow is now on `pull_request_target` it won't run until we've actually merged this PR. I tried the changes out [in a test repo](https://github.com/sentrivana/playground/pull/5).

Supersedes https://github.com/getsentry/sentry-python/pull/2493

### CODE REVIEW

**Comment by sentrivana** [comment]
cc @asottile-sentry @mdtro in case you guys want to/have time to double check

File: .github/workflows/scripts/trigger_tests_on_label.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+import argparse
+import json
+import os
+from urllib.parse import quote
+from urllib.request import Request, urlopen
+
+LABEL = "Trigger: tests"
```
Comment (asottile-sentry) [inline .github/workflows/scripts/trigger_tests_on_label.py:8]: I would maybe make this more explicit that it's sensitive tests requiring secrets

**Review by asottile-sentry — APPROVED** [review]
otherwise looks good

File: .github/workflows/scripts/trigger_tests_on_label.py (hunk 1 above)
Comment (sentrivana) [reply .github/workflows/scripts/trigger_tests_on_label.py]: Changed to `"Trigger: tests using secrets"`.

File: scripts/split-tox-gh-actions/templates/base.jinja [hunk 2]
```diff
@@ -16,6 +24,10 @@ concurrency:
 
 permissions:
   contents: read
+  {% if needs_github_secrets %}
+  # `write` is needed to remove the `Trigger: tests` label
```
Comment (antonpirker) [inline scripts/split-tox-gh-actions/templates/base.jinja:28]:
```suggestion
  # `write` is needed to remove the `Trigger: tests using secrets` label
```

**Review by antonpirker — APPROVED** [review]

**Pull request merged**